doc/plugin_server_datastore_sql.md

```diff
@@ -0,0 +1,12 @@
+# Server plugin: DataStore "sql"
+
+The `sql` plugin implements a sql based storage option for SPIRE server by using the [Gorm ORM](http://gorm.io/docs/connecting_to_the_database.html) for SQLite, MySQL and PostgreSQL database. 
+
+The plugin accepts the following configuration options based on the [Gorm ORM](http://gorm.io/docs/connecting_to_the_database.html).
+
+| Configuration     | Description                                    |
+| ------------------| ---------------------------------------------- |
+| database_type     | Gorm database type                             |
+| connection_string | Gorm connection string                         |
```

Comment:
It makes me a little uneasy to allow the underlying orm shine through here... I thought maybe it would be better to enumerate the types supported rather than refer to gorm docs. This is especially true in the event that gorm picks up support for a new dialect but SPIRE didn't get its dependency version bump yet. We already have a switch on the type value anyways.

I suppose if we did this, we would also need to add a section to this doc about the connect string format for the supported backends.

Reply:
I tried this branch out with MySQL but it looks like it ran into a problem while trying to perform the initial migration. Did everything work OK for you on Postgres? Maybe another reason to have more control over which types we allow - it may take some time to introduce support for all the types if there are specific idiosyncrasies that need to be addressed

Reply:
+1 on being specific here.

Reply:
I removed any mentions of grom from the documentation and extended the documentation a bit. 

Furthermore I removed mysql support since there were some issue. But I added a e2e test for postgres.